x/payment/pfd.go

```diff
@@ -0,0 +1,97 @@
+package payment
+
+import (
+	"context"
+	"google.golang.org/grpc"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdk_tx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+
+	"github.com/celestiaorg/celestia-app/x/payment/types"
+	"github.com/celestiaorg/nmt/namespace"
+)
+
+var (
+	// shareSizes includes all the possible share sizes of the given data
+	// that the signer must sign over.
+	shareSizes = []uint64{16, 32, 64, 128}
+)
```

Comment:
would prefer not to hardcode these, but I also think its ok to merge as is, considering that we have #239 and #236